Source/WebCore/css/StyleColor.h

```diff
@@ -31,31 +30,119 @@
 
 #pragma once
 
-#include "CSSValueKeywords.h"
 #include "Color.h"
+#include "ColorInterpolationMethod.h"
+#include "CSSValueKeywords.h"
+#include <vector>
 #include <wtf/OptionSet.h>
 
 namespace WebCore {
 
+class CSSPrimitiveValue;
+
 enum class StyleColorOptions : uint8_t {
     ForVisitedLink = 1 << 0,
     UseSystemAppearance = 1 << 1,
     UseDarkAppearance = 1 << 2,
     UseElevatedUserInterfaceLevel = 1 << 3
 };
 
-struct StyleColor {
+class StyleColor;
+
+struct CurrentColor {
+    bool operator==(CurrentColor const&) const = default;
+    static StyleColor make();
+};
+
+struct SystemColor {
+    bool operator==(SystemColor const&) const = default;
+};
+struct ColorMixComponent;
+
+struct ColorContext {
+    std::optional<Color> currentColor = std::nullopt;
+};
+
+struct ColorMixPercentages {
```

Comment:
I would take the opportunity to put this and the other ColorMix related structs / functions in their own ColorMix.h/cpp